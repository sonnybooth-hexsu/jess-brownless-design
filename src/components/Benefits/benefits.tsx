import React from "react"
import { Link } from "gatsby"
import { CheckCircle } from "react-feather"
import buttons from "../../styles/buttons.module.css"

export const Benefits = ({ benefits }) => (
  <div className="grid grid-cols-12 gap-4 lg:gap-6">
    <h2 className="col-span-12 mb-8 md:text-center md:col-span-8 md:col-start-3 xl:col-span-6 xl:col-start-4">
      Bespoke cosmetic injectables and skin treatment plans.
    </h2>
    <div className="col-span-12 mb-4 md:flex md:flex-wrap md:justify-center lg:col-start-3 lg:col-span-8">
      {benefits?.map(({ text }, id) => (
        <div
          key={id}
          data-testid="benefitsItem"
          className="flex items-center mb-4 md:mx-4 md:justify-center"
        >
          <CheckCircle size="32" className="text-gray-100" />
          <p className="pb-0 ml-4">{text}</p>
        </div>
      ))}
    </div>
    <div className="col-span-12 text-center">
      <Link
        className={`${buttons.btn} ${buttons.btnSecondary} block w-full md:inline-block md:w-auto`}
        to="/treatments"
      >
        View treatments
      </Link>
    </div>
  </div>
)

export default Benefits
