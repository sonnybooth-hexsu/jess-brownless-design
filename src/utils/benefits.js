export const Benefits = [
  {
    text: "Benefit 1",
  },
  {
    text: "Benefit 2",
  },
  {
    text: "Benefit 3",
  },
]

export default Benefits
