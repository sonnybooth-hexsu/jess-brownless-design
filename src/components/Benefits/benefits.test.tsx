/* eslint no-undef: 0 */

import React from "react"
import { render, cleanup } from "@testing-library/react"
import Benefits from "./benefits"
import benefits from "../../utils/benefits"

afterEach(() => {
  cleanup()
})

describe("<Benefits />", () => {
  it("should match snapshot", () => {
    const { container } = render(<Benefits benefits={benefits} />)

    expect(container).toMatchSnapshot()
  })

  it("has partnership image", () => {
    const { queryAllByTestId } = render(<Benefits benefits={benefits} />)

    const images = queryAllByTestId("benefitsItem")

    expect(images).toHaveLength(benefits.length)
  })
})
