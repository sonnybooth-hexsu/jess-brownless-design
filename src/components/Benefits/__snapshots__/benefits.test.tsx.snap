// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Benefits /> should match snapshot 1`] = `
<div>
  <div
    class="grid grid-cols-12 gap-4 lg:gap-6"
  >
    <h2
      class="col-span-12 mb-8 md:text-center md:col-span-8 md:col-start-3 xl:col-span-6 xl:col-start-4"
    >
      Bespoke cosmetic injectables and skin treatment plans.
    </h2>
    <div
      class="col-span-12 mb-4 md:flex md:flex-wrap md:justify-center lg:col-start-3 lg:col-span-8"
    >
      <div
        class="flex items-center mb-4 md:mx-4 md:justify-center"
        data-testid="benefitsItem"
      >
        <svg
          class="text-gray-100"
          fill="none"
          height="32"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
          />
          <polyline
            points="22 4 12 14.01 9 11.01"
          />
        </svg>
        <p
          class="pb-0 ml-4"
        >
          Benefit 1
        </p>
      </div>
      <div
        class="flex items-center mb-4 md:mx-4 md:justify-center"
        data-testid="benefitsItem"
      >
        <svg
          class="text-gray-100"
          fill="none"
          height="32"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
          />
          <polyline
            points="22 4 12 14.01 9 11.01"
          />
        </svg>
        <p
          class="pb-0 ml-4"
        >
          Benefit 2
        </p>
      </div>
      <div
        class="flex items-center mb-4 md:mx-4 md:justify-center"
        data-testid="benefitsItem"
      >
        <svg
          class="text-gray-100"
          fill="none"
          height="32"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
          />
          <polyline
            points="22 4 12 14.01 9 11.01"
          />
        </svg>
        <p
          class="pb-0 ml-4"
        >
          Benefit 3
        </p>
      </div>
    </div>
    <div
      class="col-span-12 text-center"
    >
      <a
        class="btn btnSecondary block w-full md:inline-block md:w-auto"
        href="/treatments"
      >
        View treatments
      </a>
    </div>
  </div>
</div>
`;
